llm_bioinformatics_research/src/UserProfile/UserProfile.js

```diff
@@ -0,0 +1,152 @@
+        >
+          <ManageAccounts /> My Profile
+        </Button>
+
+        <Button
+          onClick={() => setSection('settings')}
+          className={section === 'settings' ? 'selected-button' : 'default-button'} 
+        >
+          <Settings /> Settings
+        </Button>
+
+        <Button onClick={handleLogout} className="logout-button" aria-label='logoutbutton'>
+          <Logout /> Logout
+        </Button>
+      </Box>
+
+      {section === 'profile' && (
+        <Box className="profile-section">
+          <Box className="avatar-container">
+            <Avatar alt="User Image" src={user.image} className="avatar" />
+            <IconButton className="edit-icon" component="label">
+              <EditIcon />
+              <input type="file" hidden accept="image/*" onChange={handleImageChange} data-testid ='avatar-upload' />
+            </IconButton>
+          </Box>
+
+          <TextField
+            label="Name"
+            name="name"
+            value={user.name}
+            onChange={handleInputChange}
+            className="input-field"
+            margin="normal"
+          />
+          <TextField
+            label="Email"
+            name="email"
+            value={user.email}
+            onChange={handleInputChange}
+            className="input-field"
+            margin="normal"
+          />
+          <TextField
+            label="Phone Number"
+            name="phone"
+            value={user.phone}
+            onChange={handleInputChange}
+            className="input-field"
+            margin="normal"
+          />
+          <TextField
+            label="Location"
+            name="location"
+            value={user.location}
+            onChange={handleInputChange}
+            className="input-field"
+            margin="normal"
+          />
+        </Box>
+      )}
+
+      {section === 'settings' && (
+        <Box className="settings-section">
+          <FormControl className="input-field" margin="normal">
+            <InputLabel>Theme</InputLabel>
+            <Select
+              name="theme"
+              value={user.theme}
+              onChange={handleInputChange}
+            >
+              <MenuItem value="light">Light</MenuItem>
+              <MenuItem value="dark">Dark</MenuItem>
+              <MenuItem value="system">System Default</MenuItem>
+            </Select>
+          </FormControl>
+          <FormControl className="input-field" margin="normal">
+            <InputLabel>Language</InputLabel>
+            <Select
+              name="language"
+              value={user.language}
+              onChange={handleInputChange}
+            >
+              <MenuItem value="en">English</MenuItem>
+              <MenuItem value="es">Spanish</MenuItem>
+              <MenuItem value="fr">French</MenuItem>
+            </Select>
+          </FormControl>
+        </Box>
+      )}
+    </Box>
+  );
+};
+
+export default UserProfile;
```

Comment:
Overall the code looks good and is easy to understand! I pointed out some UI elements that need to be revised a little and gave some suggestions for the unit testing!